packages/flutter_tools/lib/src/commands/attach.dart

```diff
@@ -499,6 +499,12 @@ class MDnsObservatoryDiscovery {
         }
       }
       return MDnsObservatoryDiscoveryResult(srv.first.port, authCode);
+    } on SocketException catch (error) {
+      throwToolExit(
```

Comment:
I don't think this is right.  Higher up in the chain, we just assume that if this call fails we should continue trying to use other strategies to try to find the port.

Perhaps in this case we should be printing something about why it failed, but let the process continue?

Reply:
Possible, I'll dig more into this and re-open